responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -431,11 +493,28 @@ def _validate_rai_insights_input_parameters(
             if feature_metadata is not None:
                 if not isinstance(feature_metadata, FeatureMetadata):
```

Comment:
This entire branch of validations should be moved under a function in this file or maybe moved into feature_metadata.py.

Reply:
Feature metadata is not award of the task type.

Agreed on moving to separate functions, but that also applies to almost the entire validation method. I suggest doing that in a separate PR to avoid unreadable sets of changes.

Reply:
I've moved this particular block into its own method, but we should probably do more of that in follow-up PRs. Will make a note of that...